crates/bevy_math/src/cubic_splines.rs

```diff
@@ -269,30 +282,350 @@ impl<P: Point> CubicBSpline<P> {
+            [-1.0, 3.0, -3.0, 1.0],
         ];
 
+        char_matrix
+            .iter_mut()
+            .for_each(|r| r.iter_mut().for_each(|c| *c /= 6.0));
+
         let segments = self
             .control_points
             .windows(4)
-            .map(|p| CubicCurve::coefficients([p[0], p[1], p[2], p[3]], 1.0 / 6.0, char_matrix))
+            .map(|p| CubicSegment::coefficients([p[0], p[1], p[2], p[3]], char_matrix))
             .collect();
 
         CubicCurve { segments }
     }
 }
 
-/// Implement this on cubic splines that can generate a curve from their spline parameters.
+/// Error during construction of [`CubicNurbs`]
+#[derive(Debug, Error)]
+pub enum CubicNurbsError {
+    /// Provided the wrong number of knots.
+    #[error("Wrong number of knots: expected {expected}, provided {provided}")]
+    KnotsNumberMismatch {
+        /// Expected number of knots
+        expected: usize,
+        /// Provided number of knots
+        provided: usize,
+    },
+    /// The provided knots had a descending knot pair. Subsequent knots must
+    /// either increase or stay the same.
+    #[error("Invalid knots: contains descending knot pair")]
+    DescendingKnots,
+    /// The provided knots were all equal. Knots must contain at least one increasing pair.
+    #[error("Invalid knots: all knots are equal")]
+    ConstantKnots,
+    /// Provided a different number of weights and control points.
+    #[error("Incorrect number of weights: expected {expected}, provided {provided}")]
+    WeightsNumberMismatch {
+        /// Expected number of weights
+        expected: usize,
+        /// Provided number of weights
+        provided: usize,
+    },
+    /// The number of control points provided is less than 4.
+    #[error("Not enough control points, at least 4 are required, {provided} were provided")]
+    NotEnoughControlPoints {
+        /// The number of control points provided
+        provided: usize,
+    },
+}
+
+/// Non-uniform Rational B-Splines (NURBS) are powerful generalization of the [`CubicBSpline`] which can
+/// represent a much more diverse class of curves (like perfect circles and ellipses).
+///
+/// ### Non-uniformity
+/// The 'NU' part of NURBS stands for "Non-Uniform". This has to do with a parameter called 'knots'.
+/// The knots are a non-decreasing sequence of floating point numbers. The first and last three pairs of
+/// knots control the behavior of the curve as it approaches it's endpoints. The intermediate pairs
+/// each control the length of one segment of the curve. Multiple repeated knot values are called
+/// "knot multiplicity". Knot multiplicity in the intermediate knots causes a "zero-length" segments,
+/// and can create sharp corners.
+///
+/// ### Rationality
+/// The 'R' part of NURBS stands for "Rational". This has to do with NURBS allowing each control point to
+/// be assigned a weighting, which controls how much it effects the curve compared to the other points.
```

Comment:
"effects" is a noun, "affects" is a verb
```suggestion
/// be assigned a weighting, which controls how much it affects the curve compared to the other points.
```